types/underscore/index.d.ts

```diff
@@ -154,6 +154,21 @@ declare module _ {
         : TItem
         : T;
 
+    // if T is an inferrable pair, the value type for the pair
+    // if T is a list, assume that it contains pairs of some type, so any
+    // if T isn't a list, there's no way that it can provide pairs, so never
+    type PairValue<T> =
+        T extends [EnumerableKey, infer TValue, ...any[]] ? TValue
```

Comment:
Oh, darn, I thought adding `...any[]` would be a nice way to also support tuples that happen to be longer than exactly two items, but that's a TS3.0 feature so technically I should bump the target TS version if I include this. Any preference on bumping the minimum vs. removing this @jgonggrijp?

Reply:
Either is fine by me. I think passing tuples with arity >2 to `_.object` is rather exotic.

Reply:
I agree, I just figured I'd toss this in since it didn't seem like behavior I should disallow. Per [this comment](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46306#discussion_r460690504), though, I'm going to leave this out for now and add it back later.